devito/types/equation.py

```diff
@@ -225,18 +227,23 @@ def solve(eq, target, **kwargs):
         Symbolic optimizations applied while rearranging the equation. For more
         information. refer to ``sympy.solve.__doc__``.
     """
-    # Enforce certain parameters to values that are known to guarantee a quick
-    # turnaround time
-    kwargs['rational'] = False  # Avoid float indices
-    kwargs['simplify'] = False  # Do not attempt premature optimisation
-    kwargs['manual'] = True  # Force sympy to solve one line at a time for VectorFunction
     if isinstance(eq, Eq):
         eq = eq.lhs - eq.rhs if eq.rhs != 0 else eq.lhs
-    sol = sympy.solve(eq.evaluate, target.evaluate, **kwargs)[0]
-
+    sols = []
+    for e, t in zip(as_tuple(eq), as_tuple(target)):
+        # Try first linear solver
+        try:
+            cc = linear_coeffs(e.evaluate, t)
+            sols.append(-cc[1]/cc[0])
+        except ValueError:
+            warning("Equation is not affine w.r.t the target, falling back to standard"
```

Comment:
"Equation is" -- could be a whole system of equations, so perhaps just "input" ?

Reply:
It's inside the loop so single equation, for a system of equation it's still solved separately per line, we don't support LA solvers (but maybe should put it somewhere).